src/ott/geometry/costs.py

```diff
@@ -126,6 +126,7 @@ class TICost(CostFn):
   strictly convex, as well as provide the Legendre transform of :math:`h`,
   whose gradient is necessarily the inverse of the gradient of :math:`h`.
   """
+  p = 1.0
```

Comment:
I understand why you did this, but `TICost` doesn't really need `p` param, can you please revert?

Reply:
revert by moving `tree_flatten` methods as before in duplicated classes?

Reply:
removed p